src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3847,6 +3879,46 @@ static bool hasInitializer(SourcePropertySymbol property)
                     return property.DeclaredBackingField?.HasInitializer == true;
                 }
             }
+
+            static void mergePartialConstructors(ArrayBuilder<Symbol> nonTypeMembers, SourceConstructorSymbol currentConstructor, SourceConstructorSymbol prevConstructor, BindingDiagnosticBag diagnostics)
+            {
+                if (currentConstructor.IsPartialImplementation &&
+                    (prevConstructor.IsPartialImplementation || (prevConstructor.OtherPartOfPartial is { } otherImplementation && (object)otherImplementation != currentConstructor)))
+                {
+                    // A partial constructor may not have multiple implementing declarations
+                    diagnostics.Add(ErrorCode.ERR_PartialMemberDuplicateImplementation, currentConstructor.GetFirstLocation(), currentConstructor);
+                }
+                else if (currentConstructor.IsPartialDefinition &&
+                    (prevConstructor.IsPartialDefinition || (prevConstructor.OtherPartOfPartial is { } otherDefinition && (object)otherDefinition != currentConstructor)))
+                {
+                    // A partial constructor may not have multiple defining declarations
+                    diagnostics.Add(ErrorCode.ERR_PartialMemberDuplicateImplementation, currentConstructor.GetFirstLocation(), currentConstructor);
```

Comment:
>ERR_PartialMemberDuplicateImplementation [](http://example.com/codeflow?start=46&length=40)

Should this be `ERR_PartialMemberDuplicateDefinition`?

Reply:
Yes, thanks!